### DISCUSSION TITLE
multiple schema implementation

### DISCUSSION THREAD

**Discussion by patricklx** [mikro-orm/mikro-orm#1460] [General]
Hi @Aeolun 
I see you also created a PR for multiple schemas. But with another approach.
I created one here: https://github.com/mikro-orm/mikro-orm/pull/1001

In our project we needed to have the same entities in multiple schemas (so, multitenancy, same things but different schema)
Would that be also possible with your approach?

In my PR i set the schema for the whole ORM. We also have entities which belong to a specific schema and for those we have a separate orm which uses other entities.

The disadvantage is that you cannot commit to both schemas within one transaction. Is that needed for your case?

**Comment by Aeolun** [comment]
Hi @patricklx,

Yeah, our database schema is basically spread out over an entire server, so we need to be able to join/transact across all the tables in the whole server.

It's not ideal, but I'm stuck with that legacy situation.

I don't think it'd be possible to have the same entities in multiple schemas with my PR. Or at least, that'd be a different thing from what I'm attempting. That might also explain why I had no idea what was going on in your PR 😄.

Kind regards,
Bart

**Comment by patricklx** [comment]
you think we can maybe have both? If there is a schema specified on the Entity, use that one. If not use the schema defined on the connection

**Reply by Aeolun** [reply]
I guess that is how it will work regardless, since it'll fall back to whatever is set on the connection if nothing is defined in the entity already.
